lib/propmangle.js

```diff
@@ -205,6 +254,12 @@ function mangle_properties(ast, options) {
             do {
                 mangled = base54(++cache.cname);
             } while (!can_mangle(mangled));
+            
+            // HACK: to avoid mangled global references colliding with local variable names, use
+            // a prefix on all mangled global names to effectively move them to a different namespace.
+            if (mangle_globals && name in cache.global_defs)
+                mangled = "g_" + mangled;
```

Comment:
Not crazy about this hack. Collisions can be avoided in a similar fashion to the other PR you wrote.